capellambse/helpers.py

```diff
@@ -341,52 +347,40 @@ def word_wrap(text: str, width: float | int) -> list[str]:
         A list of strings, one for each line, after wrapping.
     """
 
-    def rejoin(words: cabc.Iterable[str], start: int, stop: int | None) -> str:
-        return " ".join(itertools.islice(words, start, stop))
-
-    def splitline(line: str) -> list[str]:
-        match = re.search(r"^\s*", line)
-        assert match is not None
+    def split_into_lines(line: str, width: float | int) -> list[str]:
         words = line.split()
-
-        if words:
-            words[0] = match.group(0) + words[0]
-        return words
-
-    output_lines = []
-    input_lines = collections.deque(text.splitlines())
-    while input_lines:
-        words = collections.deque(splitline(input_lines.popleft()))
         if not words:
-            output_lines.append("")
-            continue
-
-        words_count = len(words)
-        while (
-            extent_func(rejoin(words, 0, words_count))[0] > width
-            and words_count > 0
-        ):
-            words_count -= 1
+            return [line]
+
+        current_line = ""
+        lines = []
+        for word in words:
+            test_line = word if not current_line else f"{current_line} {word}"
+            if extent_func(test_line)[0] <= width:
+                current_line = test_line
+            else:
+                if current_line:
+                    lines.append(current_line)
+                current_line = word
+        if current_line:
+            lines.append(current_line)
+        return lines
 
-        if words_count > 0:
-            output_lines.append(rejoin(words, 0, words_count))
-            if words_count < len(words):
-                input_lines.appendleft(rejoin(words, words_count, None))
+    def keep_whitespace(i: int, line: str) -> bool:
+        special_chars = {"•", "-"}
+        return i == 0 or (line.lstrip() or " ")[0] in special_chars
```

Comment:
We can make `special_chars` a tuple and then simply use `line.lstrip().startswith(special_chars)`.